src/gui/qgsrelationeditorwidget.cpp

```diff
@@ -467,6 +589,186 @@ void QgsRelationEditorWidget::unsetMapTool()
+    QgsFeature featureChild;
+    while ( featureIterator.nextFeature( featureChild ) )
+    {
+      if ( mNmRelation.isValid() )
+      {
+        const QgsFeatureRequest requestFinalChild = mNmRelation.getReferencedFeatureRequest( featureChild );
+        QgsFeatureIterator featureIteratorFinalChild = mNmRelation.referencedLayer()->getFeatures( requestFinalChild );
+        QgsFeature featureChildChild;
+        while ( featureIteratorFinalChild.nextFeature( featureChildChild ) )
+        {
+          QTreeWidgetItem *treeWidgetItemChild = new QTreeWidgetItem( treeWidgetItem );
+          treeWidgetItemChild->setData( 0, static_cast<int>( MultiEditTreeWidgetRole::FeatureType ), static_cast<int>( MultiEditFeatureType::Child ) );
+          treeWidgetItemChild->setData( 0, static_cast<int>( MultiEditTreeWidgetRole::FeatureId ), featureChildChild.id() );
+          treeWidgetItemChild->setText( 0, QgsVectorLayerUtils::getFeatureDisplayString( mNmRelation.referencedLayer(), featureChildChild ) );
+          treeWidgetItemChild->setIcon( 0, QgsIconUtils::iconForLayer( mNmRelation.referencedLayer() ) );
+
+          treeWidgetItem->addChild( treeWidgetItemChild );
+
+          featureIdsMixedValues.insert( featureChildChild.id() );
+          multimapChildFeatures.insert( treeWidgetItem, featureChildChild.id() );
+        }
+      }
+      else
+      {
+        QTreeWidgetItem *treeWidgetItemChild = new QTreeWidgetItem( treeWidgetItem );
+        treeWidgetItemChild->setData( 0, static_cast<int>( MultiEditTreeWidgetRole::FeatureType ), static_cast<int>( MultiEditFeatureType::Child ) );
+        treeWidgetItemChild->setData( 0, static_cast<int>( MultiEditTreeWidgetRole::FeatureId ), featureChild.id() );
+        treeWidgetItemChild->setText( 0, QgsVectorLayerUtils::getFeatureDisplayString( mRelation.referencingLayer(), featureChild ) );
+        treeWidgetItemChild->setIcon( 0, QgsIconUtils::iconForLayer( mRelation.referencingLayer() ) );
+        treeWidgetItem->addChild( treeWidgetItemChild );
+
+        featureIdsMixedValues.insert( featureChild.id() );
+      }
+    }
+
+    treeWidgetItem->setExpanded( true );
+    mMultiEditTreeWidget->addTopLevelItem( treeWidgetItem );
+  }
+
+  // Check for mixed values
+  if ( mNmRelation.isValid() )
+  {
+    QgsFeatureIds featureIdsNotMixed;
+    for ( const QgsFeatureId &featureId : featureIdsMixedValues )
+    {
+      bool mixedValues = false;
+      for ( QTreeWidgetItem *parentTreeWidgetItem : parentTreeWidgetItems )
+      {
+        if ( ! multimapChildFeatures.values( parentTreeWidgetItem ).contains( featureId ) )
+        {
+          mixedValues = true;
+          break;
+        }
+      }
+
+      if ( !mixedValues )
+        featureIdsNotMixed.insert( featureId );
+
+
+    }
+
+    featureIdsMixedValues -= featureIdsNotMixed;
+  }
+  else
+  {
+    for ( const QgsFeatureId &featureIdJustAdded : std::as_const( mMultiEdit1NJustAddedIds ) )
```

Comment:
If I understand well, if you have just added features, you know they are related, and so you are in a green state (all items are considered to have the same value), but if you just close the dialog and reopen it, you lose this information, and so you are in orange state.

I understand why we do that but I'm not sure this is great in terms of UI.

Reply:
I agree that it is not very optimal. But for 1:n relations it is not possible to have this information without additional data or some sort of project configuration like "if creation time of feature is inside 500ms consider them as related".
But it feels quite of very edge configuration that no one will use.
Maybe it would be more consistent to just display the green state when there are no related features and orange for all other states

Reply:
> Maybe it would be more consistent to just display the green state when there are no related features and orange for all other states

You mean green when they ARE related, and orange for all other states (including 1:n relations when you have no clue if they are related or not) ? 

I think it will be better

Reply:
Oh yes I stated it wrongly... Yes I meant like this:
- Green:
  - **n:m** and **1:n**: 0 child features available
  - **n:m** with no mixed values
- Orange:
  - **n:m** with mixed values
  - **1:n** always including when we pseudo know that feature are related (just added feature)

Reply:
:+1: 
Seems ok to me, and maybe copy/paste this explanation in the code so dev would know why & when we pick orange or green

Reply:
It is know implemented that way. But I have keep the multiselection for just added features. I think if someone adds by mistake a new child feature to hundreds of parent feature he have a chance to remediate quickly and re-delete them all at once.

Reply:
Fair point!